src/Mono.Android/Java.Lang/Object.cs

```diff
@@ -91,21 +93,29 @@ internal void SetHandleOnDeserialized (StreamingContext context)
 		}
 
 #if JAVA_INTEROP
+		[DebuggerBrowsable (DebuggerBrowsableState.Never)]
+		[EditorBrowsable (EditorBrowsableState.Never)]
 		public int JniIdentityHashCode {
```

Comment:
I would think that when debugging you *would* want *something* to designate "object identity", e.g. to easily answer "are these two variables referring to the same instance"?

Is this (still?) a desirable/useful feature?  If so, what property would best be expose to the debugger?  `JniIdentityHashCode` is useful if there are *separate `Java.Lang.Object` instances wrapping the same Java instance, while `Handle` and `PeerReference` are good for seeing distinct wrappers.

Reply:
Agree, but just wondering whether that couldn't be implemented in VS debugger [the same way it is for .NET objects](https://stackoverflow.com/a/4251470)? This way we wouldn't have to check complex ids and hashes and it would feel more _native_ to devs imo.

Reply:
I didn't know that was a feature that existed.

That said, I assume that **Make Object ID** only works for identical instances.  If you have two different C# instances which wrap the same Java instance -- presumably a rare but still possible scenario -- then Make Object ID wouldn't help.  `JniIdentityHashCode` *could*.

Reply:
Well that's not exactly what I mean... I mean if there could be added a new similar functionality, probably another context menu option specialized for this usecase, something like _Make Java object ID_  that wouldn't care about the .NET instance but about the native Java instance probably using the mentioned `JniIdentityHashCode` property. That would make looking for Java instance equality much easier compared to writing down some hash codes or so. I've heard something that in .NET 5 there should me a more native interoperability with Java (and I believe that Java.Interop is the thing to be used for that?) so if that's true, this change would benefit not only Xamarin.Android users.

Reply:
I suspect the primary reason .NET has `Make Object ID` is because .NET objects do not inherently have an ID and thus you cannot tell if 2 objects are the same.  We already have that information easily available with these fields.  In essence, they had to build a workaround for something we already have.

Adding `Make Java object ID` for both VS2019 and VSMac would be a decent amount of work for a different team (Android IDE team).  I suspect it will be placed in their backlog at much lower priority than the other tasks they have and probably never implemented.

As such, let's keep these properties visible to the debugger.  Overall this is still a good win by cleaning up the IntelliSense dropdown and some of the Debugger window.

Reply:
You're right, the effort wouldn't be worth the outcome especially since it's imo something devs don't do really often.